app/meteor-accounts-saml/server/saml_server.js

```diff
@@ -196,6 +220,8 @@ Accounts.registerLoginHandler(function(loginRequest) {
 			$set: {
 				// TBD this should be pushed, otherwise we're only able to SSO into a single IDP at a time
 				'services.saml': samlLogin,
+				username,
```

Comment:
is this code updating an user's username? if so, this is not the right way to do it. this will lead to many issues, since we use the `username` for many things, you need to update it everywhere.. the correct way to do it is calling [setUsername](https://github.com/RocketChat/Rocket.Chat/blob/develop/app/lib/server/functions/setUsername.js#L99)

Reply:
Have changed it